### ISSUE TITLE
Parameterized job running periodically - multiple running jobs

### ISSUE THREAD

**Issue by anderssoder** [hashicorp/nomad#2382]
### Nomad version
0.5.4

### Issue
Not sure if it could be classified as an issue or works as by design. If by design how should we handle these situations?

I've created a parameterized job that will run periodically. The job takes backups of different backends, hence why I want it parameterized. This all works and I get a dispatched parameterized periodic job which does its job. However, I also get a separate periodic job which will always fail since it lacks the parameters expected by my script. How can we have the combination of parameterized and periodic but in these cases prevent the separate periodic job?

Here's part of nomad status output showing the issue:
```
consul-snapshot                                                   batch    50        dead
consul-snapshot/dispatch-1488361764-c786ed02                      batch    50        dead
consul-snapshot/dispatch-1488361764-c786ed02/periodic-1488361800  batch    50        dead
consul-snapshot/dispatch-1488361764-c786ed02/periodic-1488362400  batch    50        dead
consul-snapshot/dispatch-1488361764-c786ed02/periodic-1488363000  batch    50        dead
consul-snapshot/dispatch-1488373847-42a7fb74                      batch    50        dead
consul-snapshot/dispatch-1488373847-42a7fb74/periodic-1488374400  batch    50        dead
consul-snapshot/dispatch-1488373847-42a7fb74/periodic-1488375000  batch    50        dead
consul-snapshot/periodic-1488361800                               batch    50        dead
consul-snapshot/periodic-1488362400                               batch    50        dead
consul-snapshot/periodic-1488363000                               batch    50        dead
consul-snapshot/periodic-1488363600                               batch    50        dead
consul-snapshot/periodic-1488364200                               batch    50        dead
consul-snapshot/periodic-1488364800                               batch    50        dead
consul-snapshot/periodic-1488365400                               batch    50        dead
```

As you see these ones are ok `consul-snapshot/dispatch-1488361764-c786ed02/periodic-1488361800` 
but then we get this `consul-snapshot/periodic-1488361800` which we don't want. To be fair, we could do without the dispatched parameterized but not periodic job also: `consul-snapshot/dispatch-1488361764-c786ed02`

A side issue which I think has already been reported(?) is that all these batch jobs end up in dead state when completed.

### Reproduction steps
* Create a job file with both Parameterized and Periodic stanzas. (see example job file below)
* nomad run jobfile
* nomad job dispatch <-meta ""> <job>

### Job file

```
job "consul-snapshot" {
        ...
        type = "batch"
        ...
        meta {
                <meta1> = ""
                <meta2> = ""
                <meta3> = ""
        }
        parameterized {
                payload = "optional"
                meta_required = ["meta1", "meta2"]
                meta_optional = ["meta3"]
        }
        periodic {
                // Run every 10 minutes
                cron = "*/10 * * * * *"
                prohibit_overlap = true
        }
        group "periodic_jobs" {
                count = 1
                restart {...}

                task "job" {
                        driver = "docker"
                        config {
                                image = ".."
                        }
                        env {
                                ...
                        }
                }
        }
}
```

**Comment by dadgar** [comment]
Hey,

Gonna get this fix in 0.5.5 RC2. Have to say, this was a use case I never imagined when building this feature! So hats off on being creative!

-Alex Dadgar

**Comment by anderssoder** [comment]
Cheers @dadgar ! Hats off for fixing it so quick!
